include/layers/FCLayer.hpp

```diff
@@ -0,0 +1,124 @@
+#pragma once
+#include <stdexcept>
+#include <vector>
+
+template <typename ValueType>
+std::vector<ValueType> mat_vec_mul(
+    const std::vector<std::vector<ValueType> >& mat,
+    const std::vector<ValueType>& vec) {
+  std::vector<ValueType> res(mat.size());
+  ValueType elem;
+  for (size_t i = 0; i < mat.size(); i++) {
+    elem = ValueType(0);
+    for (size_t j = 0; j < vec.size(); j++) {
+      elem += mat[i][j] * vec[j];
+    }
+    res[i] = elem;
+  }
+  return res;
+}
+
+template <typename ValueType>
+class Layer {
+ public:
+  virtual std::vector<ValueType> run(const std::vector<ValueType>& input) = 0;
+};
+
+template <typename ValueType>
+class FCLayer : public Layer<ValueType> {
+ public:
+  FCLayer() : weights(), bias() {
+    inputSize = 0;
+    outputSize = 0;
+  }
```

Comment:
You can move everything to intializer list

Reply:
Ok